doc/proposals/sdk-integration-with-olm.md

```diff
@@ -75,14 +88,110 @@ Initial PR: https://github.com/operator-framework/operator-sdk/pull/1912
 
 #### OLM resources
 
-OLM understands `operator-registry` servers and served data through several objects. A [`CatalogSource`][olm-catalogsource] specifies how to communicate with a registry server. A [`Subscription`][olm-subscription] links a particular CSV channel to a `CatalogSource`, indicating from which `CatalogSource` OLM should pull an Operator. Another OLM resource that _may_ be required is an [`OperatorGroup`][olm-operatorgroup], which provides Operator namespacing information to OLM; OLM creates two `OperatorGroup`'s by default, one of which can be used for globally scoped Operators.
+OLM understands `operator-registry` servers and served data through several objects. A [`CatalogSource`][olm-catalogsource] specifies how to communicate with a registry server. A [`Subscription`][olm-subscription] links a particular CSV channel to a `CatalogSource`, indicating from which `CatalogSource` OLM should pull an Operator. Another OLM resource that _may_ be required is an [`OperatorGroup`][olm-operatorgroup], which provides Operator namespacing information to OLM. OLM creates a globally-scoped `OperatorGroup` by default, which can be used for globally-scoped Operators.
 
 These resources can be created from bundle data with minimal user input. They can also be created from manifests defined by the user; however, the SDK cannot make guarantees that user-defined manifests will work as expected.
 
+#### OperatorGroups and tenancy requirements
+
+[`OperatorGroup`][olm-operatorgroup]'s configure CSV tenancy in multiple
+namespaces in a cluster. Each Operator must be a
+[member][olm-operatorgroup-membership] with one `OperatorGroup` resource in
+the cluster, which defines a set of namespaces the CSV can exist in. A CSV's
+`installModes` determine what [type][olm-operatorgroup-installmodes] of
+`OperatorGroup` it can be a member of.
+
+No two `OperatorGroup`'s can exist in the same namespace, and a CSV with
+membership in an `OperatorGroup` of a type it does not support (determined
+by `installModes`) will transition to a failure state.
+
+Given these rules and constraints, Operator developers may have a tough time
+writing an `OperatorGroup` for their Operator initially. To assist them,
+`operator-sdk` should automate `OperatorGroup` "compilation" if one is not
+supplied.
+
+To perform compilation, the user can optionally supply the desired install
+mode type by which the CSV is installed, and the set of namespaces (may be all
+namespaces, `""`) in which the CSV will be installed. The compilation
+algorithm is as follows:
+
+```
+1. If an OperatorGroup manifest is supplied:
+    1. Use the one supplied and return.
```

Comment:
Are there use cases that will require an operator group to be supplied, or is it possible to always generate one with flags (or defaults)?

I'm wondering if we can keep it simple and cover 90% of the use cases so that the CLI flag set doesn't explode. Can we wait to see if there's demand for supplying the operator group directly?

EDIT: I kept reading. It sounds like there may be cases where this is needed for running an operator in an existing namespace that already has an operator group? If we say that that scenario is out of scope, would that simplify things? Would it kill a bunch of common use cases?

@shawn-hurley @robszumski thoughts?

Reply:
I'm not sure how common it is to roll your own `OperatorGroup`. Given info in a CSV and `--install-mode` we can create a valid `OperatorGroup` 100% of the time.